client/src/utilities/vega-specs/Bar.js

```diff
@@ -21,14 +21,33 @@ const getPaddingX = (data, maxLabelLengthX, meanPixelsPerChar, defaultPadding) =
 
 /* Calculate how much right-padding we need to add to the chart to allow the last x-axis
 ** label to render inside the bounds */
-const getRightPadding = (data, meanPixelsPerChar) => {
+const getLabelPadding = (data, meanPixelsPerChar) => {
   const horizontalPixelsPerChar = meanPixelsPerChar / 1.1; // Account for angle
   const lastValue = data[0].values[data[0].values.length - 1] || {};
   const lastLabel = lastValue.bucketValue || '';
   let lastLabelLength = lastLabel.length > 5 ? (lastLabel.length - 5) : 0;
   lastLabelLength = lastLabelLength > 27 ? 27 : lastLabelLength; // Account for truncation
+  const labelPadding = Math.floor(lastLabelLength * horizontalPixelsPerChar);
 
-  return Math.floor(lastLabelLength * horizontalPixelsPerChar);
+  return labelPadding;
+};
+
+const getLegendPadding = (data, spec, meanPixelsPerChar) => {
+  /* If there is a custom legend title, make sure we add enough padding for that too */
+  const showLegend = spec.subBucketColumn !== null;
```

Comment:
This might not be valid for ES6 anymore, but isn't it idiomatic to test for !== undefined and !== null?